libqtile/backend/x11/core.py

```diff
@@ -445,12 +445,16 @@ def update_client_lists(self) -> None:
         wids = [wid for wid, c in self.qtile.windows_map.items() if isinstance(c, window.Window)]
         self._root.set_property("_NET_CLIENT_LIST", wids)
 
-        # This needs fixing - not reflecting stacking order correctly.
-        stacked_wids = [
-            wid
-            for wid, c in self.qtile.windows_map.items()
-            if isinstance(c, window.Window) and c.group and wid in self._root.query_tree()
-        ]
+        # We rely on the stacking order from the X server
+        stacked_wids = []
+        for wid in self._root.query_tree():
+            if wid not in self.qtile.windows_map:
+                continue
+
+            win = self.qtile.windows_map[wid]
```

Comment:
By using `dict.get` only one dictionary lookup is needed, saving a bit of runtime work:
```suggestion
            win = self.qtile.windos_map.get(wid)
            if not win:
                continue
```

Reply:
This is why coding when you're tired is not the best idea.